apm-agent-plugins/apm-spring-webmvc-plugin/src/main/java/co/elastic/apm/agent/springwebmvc/SpringServiceNameInstrumentation.java

```diff
@@ -67,50 +67,47 @@ public String getAdviceClassName() {
 
     public static class SpringServiceNameAdvice {
 
-        private static final Logger logger = LoggerFactory.getLogger(SpringServiceNameAdvice.class);
-
         @Advice.OnMethodExit(suppress = Throwable.class, inline = false)
         public static void afterInitPropertySources(@Advice.This WebApplicationContext applicationContext) {
+            // avoid having two service names for a standalone jar
+            // one based on Implementation-Title and one based on spring.application.name
+            if (!ServiceInfo.autoDetected().isMultiServiceContainer()) {
```

Comment:
I think we could still honor the spring application name here in case the manifest entry is not set.
- If the manifest entry is available, we make `isMultiServiceContainer()` return `false` and then `ServiceInfo.autoDetected()` will match what was read in the manifest
- If the manifest entry is not available, we should still set it, so in the case of our test application it should still work as expected.

Reply:
Clarified this through an offline discussion: we have another fallback here for the `ServiceInfo.autoDetected()` that relies on the `.jar` name. Unfortunately when running in the IDE the test app is not packaged and thus not available.